pkg/operator/ceph/controller/spec.go

```diff
@@ -456,3 +463,52 @@ func StoredLogAndCrashVolumeMount(varLogCephDir, varLibCephCrashDir string) []v1
 		},
 	}
 }
+
+// GenerateLivenessProbeExecDaemon makes sure a daemon has a socket and that it can be called and returns 0
+func GenerateLivenessProbeExecDaemon(daemonType, daemonID string) *v1.Probe {
+	confDaemon := getDaemonConfig(daemonType, daemonID)
+
+	return &v1.Probe{
+		Handler: v1.Handler{
+			Exec: &v1.ExecAction{
+				// Run with env -i to clean env variables in the exec context
+				// This avoids conflict with the CEPH_ARGS env
+				//
+				// Example:
+				// env -i sh -c ceph --admin-daemon /run/ceph/ceph-osd.0.asok status
+				Command: []string{
+					"env",
+					"-i",
+					"sh",
+					"-c",
+					fmt.Sprintf("ceph --admin-daemon %s %s", confDaemon.buildSocketPath(), confDaemon.buildAdminSocketCommand()),
+				},
+			},
+		},
+		InitialDelaySeconds: 10,
+	}
+}
+
+func getDaemonConfig(daemonType, daemonID string) *daemonConfig {
+	return &daemonConfig{
+		daemonType: string(daemonType),
```

Comment:
Can we get rid of the `DaemonType` while we're making this change? It's just a string so I don't see why we have that type definition. Then we wouldn't need to cast this to a string.

Reply:
Hum yes